File: src/test/java/eotree/data/TestEOBytesData.java
package eotree.data;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Arrays;
import java.util.stream.Collectors;
import org.junit.jupiter.api.Test;

/**
 * EO bytes data tests.
 */
public class TestEOBytesData {

    @Test
    public void testGenerateEOZeroIndent() {
        var bytes = new EOBytesData(
                new EOByte((byte) 1),
                new EOByte((byte) 255)
        );

        assertEquals(bytes.generateEO(0), "01-FF");
    }

    @Test
    public void testGenerateEONonZeroIndent() {
        var bytes = new EOBytesData(
                Arrays.stream(new EOByte[] {
                        new EOByte((byte) 1),
                        new EOByte((byte) 255)
                }).collect(Collectors.toList())
        );

        assertEquals(bytes.generateEO(1), "01-FF");
    }
}

File: src/test/kotlin/eotree/data/TestEOBytesData.kt
package eotree.data

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.polystat.j2eo.eotree.data.EOByte
import org.polystat.j2eo.eotree.data.EOBytesData
import java.util.Arrays
import java.util.stream.Collectors

/**
 * EO bytes data tests.
 */
class TestEOBytesData {
    @Test
    fun testGenerateEOZeroIndent() {
        val bytes = EOBytesData(
            EOByte(1.toByte()),
            EOByte(255.toByte())
        )
        Assertions.assertEquals(bytes.generateEO(0), "01-FF")
    }

    @Test
    fun testGenerateEONonZeroIndent() {
        val bytes = EOBytesData(
            Arrays.stream(
                arrayOf(
                    EOByte(1.toByte()),
                    EOByte(255.toByte())
                )
            ).collect(Collectors.toList())
        )
        Assertions.assertEquals(bytes.generateEO(1), "01-FF")
    }
}
